pkg/service/publish_results.go

```diff
@@ -57,6 +58,18 @@ func decodeGRPCResultCollection(_ context.Context, grpcReq interface{}) (interfa
 	}, nil
 }
 
+func decodeJSONRPCResultCollection(_ context.Context, msg json.RawMessage) (interface{}, error) {
+	var req resultCollection
+
+	if err := json.Unmarshal(msg, &req); err != nil {
+		return nil, &jsonrpc.Error{
+			Code:    -32000,
```

Comment:
I was investigating why we use this magic number, and it's probably more correct to use `-32700`, as described in the constants for the [jsonrpc package docs](https://pkg.go.dev/github.com/go-kit/kit@v0.8.0/transport/http/jsonrpc?tab=doc). 

I'm also not sure how much we care about using the correct error codes, since we're writing both the client and the server, and `-32000` is defined in the [jsonrpc spec](https://www.jsonrpc.org/specification#error_object) as a valid application-specific server error code.

Reply:
I copy pasted that from the docs, and left it all TBD. Do you think it makes sense to return a `jsonrpc.Error` or just `errors.Wrap` here?

Reply:
I think `jsonrpc.Error` is fine. I enjoy the thought of us returning relevant jsonrpc errors and error codes, such that someone using launcher could use their own jsonrpc-compliant server (unlikely as that may be).

Reply:
I'm good with this then. I suspect there's future work in normalizing the GRPC errors and the JSON rpc ones. But it doesn't seem worth thinking hard now.